airflow/api_ui/security.py

```diff
@@ -0,0 +1,106 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Callable, cast
+
+from fastapi import Depends, HTTPException, Request
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+from typing_extensions import Annotated
+
+from airflow.auth.managers.base_auth_manager import ResourceMethod
+from airflow.auth.managers.models.base_user import BaseUser
+from airflow.auth.managers.models.resource_details import DagAccessEntity, DagDetails, DatasetDetails
+from airflow.providers.fab.auth_manager.api.auth.backend.basic_auth import auth_current_user
+from airflow.providers.fab.auth_manager.models import User
+from airflow.www.extensions.init_auth_manager import get_auth_manager
+
+security = HTTPBasic()
+
+
+def method(request: Request) -> ResourceMethod:
+    return cast(ResourceMethod, request.method)
+
+
+def check_authentication(
+    credentials: Annotated[HTTPBasicCredentials, Depends(security)],
+) -> User | None:
+    """Check that the request has valid authorization information."""
+    # TODO:
+    #    - Handle other auth backends
+    #    - Handle AUTH_ROLE_PUBLIC
```

Comment:
`AUTH_ROLE_PUBLIC` is part of FAB auth manager. You should not handle it directly. I just realized that it is not handled properly today in Airflow. We should move it to FAB auth manager

Reply:
Ok then i'll remove the comment and you guys implement it in the FAB auth manager ?

Reply:
Yep, I'll create a PR for it

Reply:
#42280